testcases/GcovSetup.py

```diff
@@ -60,15 +60,21 @@ def runTest(self):
             self.installer = "yum install"
         elif self.distro_name == 'sles':
             self.installer = "zypper install"
-        dep_packages = ["rpm-build", "yum-utils", "gcc*", "perl*", "tiny*"]
+        dep_packages = ["rpm-build", "gcc*", "perl*", "tiny*"]
+        distro = None
         log.info(f"\nInstalling following dependency packages\n {dep_packages}")
         for pkg in dep_packages:
-            self.cv_HOST.host_run_command(f"{self.installer} {pkg} -y")
+            if self.distro_name == 'rhel':
+                dep_packages.append("yum-utils")
+                self.cv_HOST.host_run_command(f"{self.installer} {pkg} -y")
+            elif self.distro_name == 'sles':
+                self.cv_HOST.host_run_command(f"{self.installer} -y {pkg}")
         log.info("\nInstalling the ditro src...")
         if self.distro_name == 'rhel':
             src_path = self.util.get_distro_src('kernel', '/root', "-bp")
         elif self.distro_name == 'sles':
-            src_path = self.util.get_distro_src('kernel-default', '/root', "-bp")
+            src_path = self.util.get_distro_src('kernel-default', '/root', "-bp", "linux")
+            distro = "sles"
```

Comment:
why not make distro a class variable,  so this will expand the scope into all methods.. and you no need to pass distro as extra parameter in each of these function  `self.kernel_config(src_path, distro)`   ` self.add_gcov_param(src_conf_file, distro)`

Reply:
yes good idea